system/system_monitor/include/system_monitor/voltage_monitor/voltage_monitor.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file voltage_monitor.h
+ * @brief  voltage monitor class
+ */
+
+#ifndef SYSTEM_MONITOR__VOLTAGE_MONITOR__VOLTAGE_MONITOR_HPP_
+#define SYSTEM_MONITOR__VOLTAGE_MONITOR__VOLTAGE_MONITOR_HPP_
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+
+#include <climits>
+#include <string>
+
+class VoltageMonitor : public rclcpp::Node
+{
+public:
+  /**
+   * @brief constructor
+   * @param [in] options Options associated with this node.
+   */
+  explicit VoltageMonitor(const rclcpp::NodeOptions & options);
+
+  /**
+   * @brief Update the diagnostic state.
+   */
+  void update();
+
+protected:
+  using DiagStatus = diagnostic_msgs::msg::DiagnosticStatus;
+
+  diagnostic_updater::Updater updater_;  //!< @brief Updater class which advertises to /diagnostics
+
+  char hostname_[HOST_NAME_MAX + 1];  //!< @brief host name
+
+  /**
+   * @brief check CMOS Battey
+   * @param [out] stat diagnostic message passed directly to diagnostic publish calls
+   * @note NOLINT syntax is needed since diagnostic_updater asks for a non-const reference
+   * to pass diagnostic message updated in this function to diagnostic publish calls.
+   */
+  void checkVoltage(
+    diagnostic_updater::DiagnosticStatusWrapper & stat);  // NOLINT(runtime/references)
+  /**
+   * @brief check CMOS Battey
+   * @param [out] stat diagnostic message passed directly to diagnostic publish calls
+   * @note NOLINT syntax is needed since diagnostic_updater asks for a non-const reference
+   * to pass diagnostic message updated in this function to diagnostic publish calls.
+   */
+  void checkBatteryStatus(
+    diagnostic_updater::DiagnosticStatusWrapper & stat);  // NOLINT(runtime/references)
+
+  bool sensors_exists_;  //!< @brief flag if sensors exists
+  float voltage_warn_;
+  float voltage_error_;
```

Comment:
This is not used for judgement of error. Please remove it if you do not intend to use it.
The parameter `cmos_battery_error` as well.

Reply:
changed sensor_exists_ to a local valiavle.

Reply:
@nobuotakamasa No, I meant `voltage_error_`, not `sensor_exists_`. Please revert that commit.

`voltage_error_` is set from ROS parameter but unused. Related to this, `cmos_battery_error` looks like unnecessary.
```cpp
voltage_error_ = declare_parameter<float>("cmos_battery_error", 2.75);
```

Reply:
Thank you.
Either way, sensors_exists_ are unnecessary.
Deleted sensors_exists_.
Fixed to use voltage_error_.